<launch>

  <arg name="model" default="phoenix"/>
  <arg name="child_frame_id" default="$(arg model)/base_link"/>
  <arg name="publish_tf" default="true"/>
  <arg name="gps_fps" default="20"/>
  <arg name="output" default="screen"/>
  <arg name="quad_namespace"/>
  <arg name="gbxport" default="5555"/>

  <node pkg="gps_imu" type="gpsImu_node" name="gpsimuFilter"
    output="$(arg output)" ns="$(arg quad_namespace)">
    <param name="child_frame_id" type="string" value="$(arg child_frame_id)"/>
    <param name="publish_tf" type="bool" value="$(arg publish_tf)"/>
    <param name="gps_fps" type="int" value="$(arg gps_fps)"/>
    <param name="quadPoseTopic" type="string" value="selfpub"/>
    <param name="quadName" type="string" value="$(arg model)"/>
    <param name="rtktopic" type="string" value="SingleBaselineRTK"/>
    <param name="a2dtopic" type="string" value="Attitude2D"/>
    <param name="viconTopic" type="string" value="phoenix/vicon"/>
    <param name="posePubTopic" type="string" value="selfpub"/>
    <param name="maxThrust" value="16.4808"/>
    <param name="useUDP" type="bool" value="false"/>

    <!--nodeType determines how to process the data and what input stream to use.
        1 = gbx
        2 = ros
        3 = vicon
        4 = rosbag
        Vicon will be implemented at a later date-->
    <param name="nodeType" type="int" value="2"/>

    <param name="gbxport" type="int" value="$(arg gbxport)"/>
    <param name="minimumTestStat" value="5"/> <!--do not process measurements with worse teststat-->
    <param name="arenaCenterX" value="-742015.09"/> <!--ECEF-->
    <param name="arenaCenterY" value="-5462218.82"/>
    <param name="arenaCenterZ" value="3198013.72"/> 
    <!--set these equal to 0, run gps_kf, monitor output, then set these parameters equal to the errors. Should be cm-level. -->
    <param name="arenaCenterX_ENU" value="-0.10"/>  <!-- -0.553-->
    <param name="arenaCenterY_ENU" value="0.005"/>  <!-- -0.538-->
    <param name="arenaCenterZ_ENU" value="-0.05"/>   <!--5.720-->
    <param name="thetaWRW" value="6.2"/> <!-- offset from ENU to WRW0 in degrees-->

    <param name="runLynx" type="bool" value="true"/>
    <param name="runSnap" type="bool" value="false"/>
  </node>

</launch>
